RIOT-OS/RIOT
# tapsetup: add more features

### Contribution description
This is a first step towards fixing by adding new functionalities to the `tapsetup` script (which I plan to use late for the common approach or at least some parts of it):

- add capability to list the bridge and all interfaces that belong to it (hopefully to replace the various `get_bridge` python implementation throughout the tests)
- add capability to activates forwarding on the bridge
- add capability to add addresses to the bridge
- add capability to add routes to the bridge



### Testing procedure
I tested this on both Linux and FreeBSD but the commands I used for FreeBSD are the same on OSX, it should also work there.



```sh
./dist/tools/tapsetup/tapsetup -c
./dist/tools/tapsetup/tapsetup -l tap0
./dist/tools/tapsetup/tapsetup -d
```

should list the bridge `tap0` belongs to and its interfaces. If you use the name of the bridge, `tapbr0` on Linux, `bridge0` on FreeBSD, it should work as well.

```sh
./dist/tools/tapsetup/tapsetup -f
```

should create a bridge with forwarding enabled

```sh
./dist/tools/tapsetup/tapsetup -f -d
```

will revert that.

```sh
./dist/tools/tapsetup/tapsetup -a 2001:db8::1/48 -a fe80::1
```

should add the addresses `2001:db8::1/48` and `fe80::1/128` to the bridge when it is created


```sh
./dist/tools/tapsetup/tapsetup -a 2001:db8::1/48 -a fe80::1 -d
```

should revert it.

```sh
./dist/tools/tapsetup/tapsetup -r 2001:db8::/48 fe80::2
```

should add a route to `2001:db8::/48` via `fe80::2` and the bridge and

```sh
./dist/tools/tapsetup/tapsetup -r 2001:db8::/48 fe80::2 -d
```

should revert it again.

You may also try different combinations of the command.




### Issues/PRs references
#14618
